### ISSUE TITLE
Realign turrets before undeploying

### ISSUE THREAD

**Issue by reaperrr** [OpenRA/OpenRA#13285]
Just like actors turn to a specific facing before deploying, their turrets should realign with facing before undeploying, to avoid visual glitches.

**Comment by anpage** [comment]
How would this affect balance? Wouldn't animating to face forward cause the undeploy process to take a little longer?

**Comment by penev92** [comment]
Yes, it would take longer to undeploy. Shouldn't be much of an issue for some visual polish though.

**Comment by anpage** [comment]
If the original did this, then I'm all for it.

I've been looking for a way to contribute, and I feel like this is a good way to get started. I'll work on a fix if nobody else is.

**Comment by pchote** [comment]
This would be a good first bug, so you're certainly welcome to have a go.

I recommend creating a new `WaitForTurretAlignment` activity (refer to `Turn` for a reasonable example), which sets `Turreted.DesiredFacing` and then waits until it reaches the desired facing before continuing.  This should be added to the activity queue from the `Transform` activity's  `OnFirstRun` method.

**Comment by GraionDilach** [comment]
> the fact that the original game already did this suggests that the (minor) balance issues would come from not fixing this.

The original games only did this if a deploy animation is present. YR IsSimpleDeployer (Siege Chopper) logic without a deployanim will not realign the turret.

**Comment by matjaeck** [comment]
`Easy` label should probably get removed when reopening per #15947.
